scripts/servers.py

```diff
@@ -220,9 +221,18 @@ def managed_firebase_auth_emulator(
     Yields:
         psutil.Process. The Firebase emulator process.
     """
+    if not constants.DEV_MODE:
+        from core.domain import platform_parameter_list
```

Comment:
This does not seem right ... we can't have scripts/ relying on core/. The former (dev scripts) is just supposed to be individual standalone things and shouldn't assume the existence of a datastore etc.

This might mean we need a project ID to be stored in common.py for dev purposes so that we can use it here. That project ID is a dummy one (dev-project-id) that is different from the one used in the codebase/production.

Reply:
From our previous discussion [here](https://github.com/oppia/release-scripts/issues/137#issuecomment-2282806872), I tried out what you suggested [here](https://github.com/oppia/release-scripts/issues/137#issuecomment-2285512292). So, `common.py` references platform params and then scripts/ access those params. But even in that case the import issue remains the same i.e. requirement of new dev dependencies. 

Are you suggesting to have the inline import in `common.py` instead? Since `common.py` is also within scripts/, why is that different though?

Reply:
There are two other feconf constants dependent on oppia_project_id:

- `ENV_IS_OPPIA_ORG_PRODUCTION_SERVER`
- `OPPIA_SITE_URL`

Another approach here can be creating a util functions in feconf or a new lib `core/project_id_conf.py` to fetch these three params which limits the import to that lib instead of adding it in scripts and introducing multiple inline imports. WDYT?

Reply:
> From our previous discussion [here](https://github.com/oppia/release-scripts/issues/137#issuecomment-2282806872), I tried out what you suggested [here](https://github.com/oppia/release-scripts/issues/137#issuecomment-2285512292). So, `common.py` references platform params and then scripts/ access those params. But even in that case the import issue remains the same i.e. requirement of new dev dependencies.
> 
> Are you suggesting to have the inline import in `common.py` instead? Since `common.py` is also within scripts/, why is that different though?

Wait, no, that's not what I meant. My thinking was that common.py in scripts should store hardcoded constants, and not refer to the platform params in the "core" code at all. Otherwise, as you say, scripts would still depend on core.

Does that help?

Reply:
Yes, that makes sense, updated now. I am just using the dev project ID in feconf though instead of hardcoding in constants. Does that sound good?

Reply:
Can common.py import feconf? If so then I think that's probably fine too, yeah.

Reply:
Hmm, feconf is already imported directly in servers in the existing codebase. So, if we want to use the hardcoded value (i.e. the dev project id set in feconf), can you guide me on why do we want to move it to common and then have servers get it from common?

Reply:
Oh, if it's already imported in servers I think that's OK.

My thinking was that scripts/ should be separate from core/, and feconf.py is in core/. So if they can't be fully separate then perhaps we can at least ensure that the only link to core/ is via scripts/common.py. But I'm fine either way really.

Reply:
Sg, keeping it as it is then.